apps/arweave/include/ar_blacklist_middleware.hrl

```diff
@@ -74,9 +80,15 @@ end).
 		[<<"vdf">>] ->
 			{get_vdf, maps:get(get_vdf, LimitByIP, 180)};
 		[<<"vdf">>, <<"session">>] ->
-			{get_vdf, maps:get(get_vdf, LimitByIP, 180)};
+			{get_vdf_session, maps:get(get_vdf_session, LimitByIP, 30)};
+		[<<"vdf2">>, <<"session">>] ->
+			{get_vdf_session, maps:get(get_vdf_session, LimitByIP, 30)};
+		[<<"vdf3">>, <<"session">>] ->
+			{get_vdf_session, maps:get(get_vdf_session, LimitByIP, 30)};
```

Comment:
Are you referring to numerous "vdf_validation_start .." messages? Those should be serial in fact, at most about two messages per second. I added a [patch](https://github.com/ArweaveTeam/arweave/pull/578/commits/82b3d9ed27e41e625ce4c777f73c34c2acbaad17) to avoid making extra session requests when they are not needed.

Reply:
Ah no, something different. Let me find the logs. Sounds like a separate issue.

Reply:
Hm, yeah we can ignore this. I remember seeing some logs yesterday where it looks like my miner was validating the same block multiple times - like it received the same block several times from the same IP (an ario node I think), and kicked off multiple concurrent validations processes for it. But I can't find the logs anymore. So let's ignore.